crates/datastore/src/traits.rs

```diff
@@ -304,16 +314,26 @@ impl TxData {
         self.truncates.iter().copied()
     }
 
-    /// Check if this [`TxData`] contains any `inserted | deleted` rows or `connect/disconnect` operations.
+    /// Check if this [`TxData`] contains any `inserted | deleted` rows
+    /// or `connect/disconnect` operations.
+    ///
+    /// Mutations of ephemeral tables are excluded, i.e. if the transaction
+    /// modifies only ephemeral tables (and is not a connect/disconnect operation),
+    /// the method returns `false`.
     ///
     /// This is used to determine if a transaction should be written to disk.
     pub fn has_rows_or_connect_disconnect(&self, reducer_context: Option<&ReducerContext>) -> bool {
-        self.inserts().any(|(_, inserted_rows)| !inserted_rows.is_empty())
-            || self.deletes().any(|(.., deleted_rows)| !deleted_rows.is_empty())
-            || matches!(
-                reducer_context.map(|rcx| rcx.name.strip_prefix("__identity_")),
-                Some(Some("connected__" | "disconnected__"))
-            )
+        let is_non_ephemeral_mutation =
+            |(table_id, rows): (_, &Arc<[_]>)| !(self.is_ephemeral_table(table_id) || rows.is_empty());
+
+        let has_inserts = self.inserts().any(is_non_ephemeral_mutation);
+        let has_deletes = self.deletes().any(is_non_ephemeral_mutation);
+        let is_connect_disconnect = matches!(
+            reducer_context.map(|rcx| rcx.name.strip_prefix("__identity_")),
+            Some(Some("connected__" | "disconnected__"))
+        );
+
+        has_inserts || has_deletes || is_connect_disconnect
```

Comment:
I'm not sure LLVM will understand making this lazy, so this may do unnecessary work beyond early returning after `has_inserts`. I'd prefer avoiding the temporaries just to make this clear to LLVM.